### PR TITLE
Feature/add client capability flow

### PR DESCRIPTION
**Pull request by CoralMalachi**
The following pull request adds support for passing Controller Certification
Client Capability Query test.
According to section 5.6.2 in the EasyMesh Test Plan v1.0 [1],
the MCUT (MultiAP Controller Under Test) sends a client capability query
in order to get information about the CTT STA.

The following changes were made :
1.     The Agent subscribes to a message of type :	CLIENT_CAPABILITY_QUERY_MESSAGE.

2.     Add handle method which responsible for handle the CMDU when receive at the agent.

3.     A shell test flow have been added to `test_flows.sh` for the Client Capability Query,
       according to section 5.6.2 in the EasyMesh Test Plan v1.0 [1].

### CODE REVIEW

**Comment by morantr** [comment]
Why commit https://github.com/prplfoundation/prplMesh/pull/188/commits/2a1736a574853ce50666b8cac5d67921c550d941 is here?

File: agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp [hunk 1]
```diff
@@ -126,8 +126,8 @@ bool main_thread::init()
             ieee1905_1::eMessageType::AP_AUTOCONFIGURATION_WSC_MESSAGE,
             ieee1905_1::eMessageType::CHANNEL_PREFERENCE_QUERY_MESSAGE,
             ieee1905_1::eMessageType::CHANNEL_SELECTION_REQUEST_MESSAGE,
-            ieee1905_1::eMessageType::ACK_MESSAGE,
-            ieee1905_1::eMessageType::TOPOLOGY_QUERY_MESSAGE})) {
+            ieee1905_1::eMessageType::ACK_MESSAGE, ieee1905_1::eMessageType::TOPOLOGY_QUERY_MESSAGE,
+            ieee1905_1::eMessageType::CLIENT_CAPABILITY_QUERY_MESSAGE})) {
```
Comment (morantr) [inline agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp:130]: Clang format did something ugly here. Is it possible to fix it?

File: agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp [hunk 2]
```diff
@@ -1582,6 +1585,18 @@ bool main_thread::handle_1905_discovery_query(ieee1905_1::CmduMessageRx &cmdu_rx
...
 
+/**
+ * @brief Handles 1905 Discovery Query message, currently only prints to the log
+ * @param cmdu_rx
+ * @return true on success
+ * @return false on failure
+ */
+bool main_thread::handle_1905_client_capability_query(ieee1905_1::CmduMessageRx &cmdu_rx)
+{
+    LOG(DEBUG) << "Received Client capability Query message";
```
Comment (morantr) [inline agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp]: Ok we received capability query. 
Don't forget to create a response? If you don't know what to fill the TLVs in the response, use dummy values.

**Review by morantr — DISMISSED** [review]

File: agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp (hunk 2 above)
Comment (tomereli) [reply agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp]: No need for a response, it is enough to just verify the request was received.

File: agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp (hunk 1 above)
Comment (CoralMalachi) [reply agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp:130]: fixed.

**Comment by CoralMalachi** [comment]
> Why commit [2a1736a](https://github.com/prplfoundation/prplMesh/commit/2a1736a574853ce50666b8cac5d67921c550d941) is here?

Because the current PR is based on this commit.

File: agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.h [hunk 3]
```diff
@@ -70,6 +70,7 @@ class main_thread : public btl::transport_socket_thread {
                                                 const std::string &src_mac);
     bool handle_1905_discovery_query(ieee1905_1::CmduMessageRx &cmdu_rx);
     //bool sta_handle_event(const std::string &iface,const std::string& event_name, void* event_obj);
+
```
Comment (tomereli) [inline agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.h]: remove accidentally added new line

File: agent/src/beerocks/slave/son_slave_thread.cpp [hunk 4]
```diff
@@ -4701,8 +4701,9 @@ bool slave_thread::parse_non_intel_join_response(Socket *sd)
 
 bool slave_thread::handle_client_capability_query(Socket *sd, ieee1905_1::CmduMessageRx &cmdu_rx)
 {
+    // LOG(DEBUG) << "Received CLIENT CAPABILITY QUERY MESSAGE";
     const auto mid = cmdu_rx.getMessageId();
-    LOG(DEBUG) << "Received CLIENT_CAPABILITY_QUERY_MESSAGE, mid=" << std::dec << int(mid);
+    LOG(DEBUG) << "Received CLIENT CAPABILITY QUERY MESSAGE , mid=" << std::dec << int(mid);
```
Comment (tomereli) [inline agent/src/beerocks/slave/son_slave_thread.cpp]: change back

File: tools/docker/tests/test_flows.sh [hunk 5]
```diff
@@ -6,7 +6,7 @@
 ###############################################################
 
 ALL_TESTS="topology initial_ap_config ap_config_renew ap_config_bss_tear_down channel_selection
-           ap_capability_info_reporting client_capability_info_reporting link_metric_collection
+          ap_capability_info_reporting client_capability_info_reporting link_metric_collection
```
Comment (tomereli) [inline tools/docker/tests/test_flows.sh]: why this change of indentation?

File: tools/docker/tests/test_flows.sh [hunk 6]
```diff
@@ -27,6 +27,7 @@ start_gw_repeater() {
 }
 
 send_bml_command() {
+    echo docker exec -it gateway ${installdir}/bin/beerocks_cli -c "$*"
```
Comment (tomereli) [inline tools/docker/tests/test_flows.sh]: no need for the echo

File: tools/docker/tests/test_flows.sh [hunk 7]
```diff
@@ -98,8 +99,12 @@ test_ap_capability_info_reporting() {
 }
 
 test_client_capability_info_reporting() {
-    #TODO: Implement
-    return 1
+#TODO: Implement
```
Comment (tomereli) [inline tools/docker/tests/test_flows.sh]: remove TODO - you're doing this

File: tools/docker/tests/test_flows.sh [hunk 8]
```diff
@@ -98,8 +99,12 @@ test_ap_capability_info_reporting() {
 }
 
 test_client_capability_info_reporting() {
-    #TODO: Implement
-    return 1
+#TODO: Implement    
+    status "test client capability"   
+    eval send_bml_command '"bml_wfa_ca_controller \"DEV_SEND_1905,DestALid,$AL_MAC,MessageTypeValue,0x8009,tlv_type,0x90,tlv_length,0xC,tlv_value,{$RADIO_WLAN0_MAC 0x000000110022}\""' $redirect
```
Comment (tomereli) [inline tools/docker/tests/test_flows.sh]: split to multiple lines

File: tools/docker/tests/test_flows.sh [hunk 9]
```diff
@@ -98,8 +99,12 @@ test_ap_capability_info_reporting() {
...
+    sleep 1
+    dbg "Confirming client capability query has been received on agent"
+    docker exec -it repeater sh -c 'grep -i -q "CLIENT CAPABILITY" /tmp/$USER/beerocks/logs/beerocks_agent_wlan0.log'
```
Comment (tomereli) [inline tools/docker/tests/test_flows.sh]: Also, check that the other agent received it. Currently, both will just print `CLIENT CAPABILITY` since they have the same handle, but in the future, we will add a check to verify if the `BSSID == radio mac`.

File: tools/docker/tests/test_flows.sh [hunk 10]
```diff
@@ -151,6 +156,8 @@ test_init() {
         exit 1
     }
     AL_MAC=$(docker exec -it gateway ${installdir}/bin/beerocks_cli -c bml_conn_map | grep IRE_BRIDGE | awk '{print $5}' | cut -d ',' -f 1)
+    RADIO_WLAN0_MAC=$(docker exec -it gateway ${installdir}/bin/beerocks_cli -c bml_conn_map | grep "RADIO: wlan0" | head -1 | awk '{print $4}' | cut -d ',' -f 1)
```
Comment (tomereli) [inline tools/docker/tests/test_flows.sh]: Add also `RADIO_WLAN2_MAC`. Might not be used for now, but can be useful for the future.

File: tools/docker/tests/test_flows.sh [hunk 11]
```diff
@@ -209,5 +216,6 @@ main() {
 
 VERBOSE=false
 AL_MAC=
+RADIO_WLAN0_MAC=
```
Comment (tomereli) [inline tools/docker/tests/test_flows.sh:219]: Add RADIO_WLAN0_MAC

**Review by tomereli — CHANGES_REQUESTED** [review]

File: agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp (hunk 1 above)
Comment (morantr) [reply agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp:130]: It doesn't seem like that...
Make sure that after you fix it and run the clang_format.sh, it doesn't change it again to the ugly form.

**Review by tomereli — APPROVED** [review]

File: agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp (hunk 1 above)
Comment (CoralMalachi) [reply agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp:130]: I have tried, Travis fails when I change the format by myself.
